src/many-client-macros/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{quote, ToTokens};
+use syn::parse2;
+use syn::FnArg;
+use syn::ItemTrait;
+use syn::TraitItemMethod;
+use syn::Type;
+use syn::{parse_macro_input, parse_quote, AttributeArgs, LitStr};
+
+fn type_and_namespace(arguments: AttributeArgs) -> (Type, Option<LitStr>) {
+    let mut r#type = None;
+    let mut lit = None;
+    for argument in arguments {
```

Comment:
I wonder if you can't just `parse2::<(Type, Option<LitStr>)>` or similar. I know you can declare your own type and `impl syn::parse::Parse for ...`. Look at how I did it in `many-macros/src/lib.rs` for the endpoint which is what you want.

There's a bug here where I could do `many_client("hello", "world", SomeStruct, "blue")`. It's not proper.